main.py

```diff
@@ -58,6 +58,9 @@ def setup_ui(self):
         self.ui.findChild(QtCore.QObject, ADD_NEW_STREAM) \
             .triggered.connect(self.add_new_stream)
 
+        self.reset_menu = self.ui.findChild(QtCore.QObject, "menuReset")
+        self.url_list = []
```

Comment:
I think this can be added into the VideoFrameGrid instead. Probably better to move as much as we can away from main.